File: cpp/Binary Tree/L_averageOfLevel.cpp
#include<bits/stdc++.h>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

vector<double> averageOfLevels(TreeNode* root) {
    queue<TreeNode*> q;
    vector<double> res;
    q.push(root);
    while (!q.empty()) {
        int n = q.size();
        double total = 0;
        for(int i=0;i<n;i++) {
            TreeNode *curr = q.front();
            q.pop();
            if(curr->left) q.push(curr->left);
            if(curr->right) q.push(curr->right);
            total += curr->val;
        }
        res.push_back(total/n);
    }
    return res;
}
int main() {

}

File: Python/Binary Trees/L_averageOfLevel.py
from collections import deque


def averageOfLevels(root):
        queue = deque([root])
        res = []
        while queue:
            n = len(queue)
            total = 0
            for i in range(n):
                curr = queue.popleft()
                if curr.left:
                    queue.append(curr.left)
                if curr.right:
                    queue.append(curr.right)
                total += curr.val
            res.append(total/n)
        return res